### PR TITLE
Add new VSI job to trigger VSI netcore tao integration tests on explicit request.

### PR DESCRIPTION
**Pull request by mavasani**
Once the tests and infrastructure are working fine, we can switch them to run for every pr commit.

Cloned and edited from https://github.com/dotnet/roslyn-internal/blob/master/netci.groovy#L143

### CODE REVIEW

**Comment by mavasani** [comment]
test ci please

**Comment by mavasani** [comment]
Tagging @dotnet/project-system @dotnet/roslyn-infrastructure @tannergooding for review.

File: netci.groovy [hunk 1]
```diff
@@ -32,4 +32,55 @@ def branch = GithubBranchName
...
+    steps {
+        // Build roslyn-project-system repo.
+        batchFile("build.cmd /release")
+
+        // git clone roslyn-internal to build and run VSI tao tests.
+        batchFile("""git clone https://github.com/dotnet/roslyn-internal.git
+pushd roslyn-internal
+git submodule init
+git submodule sync
+git submodule update --init --recursive
```
Comment (333fred) [inline netci.groovy]: Shouldn't this be all we need? `update` with `--init` and `--recursive` should do a init and sync.

**Comment by 333fred** [comment]
Can we add documentation so we know the phrase to trigger a run, or add the helper utility job so we can dotnet what it is?

File: netci.groovy (hunk 1 above)
Comment (mavasani) [inline netci.groovy]: You may be correct, though I have kept this consistent with https://github.com/dotnet/roslyn-internal/blob/master/netci.groovy#L114

**Comment by mavasani** [comment]
test ci please

**Comment by mavasani** [comment]
test ci please

**Comment by mavasani** [comment]
@333fred I have added the help job - I'll try it on an existing PR once this is merged in.

File: netci.groovy [hunk 2]
```diff
@@ -32,4 +32,60 @@ def branch = GithubBranchName
...
+def newVsiJobName = Utilities.getFullJobName(project, "vsi", false /* isPr */)
+
+def newVsiJob = job(newVsiJobName) {
+    description('')
+    label('windows-roslyn-internal-dev15-project-system')
+
+    // This opens the set of build steps that will be run.
+    steps {
+        // Build roslyn-project-system repo.
+        batchFile("build.cmd /release")
```
Comment (mavasani) [inline netci.groovy]: FYI: Currently, build.cmd always runs unit tests after build. We may want to either separate this script into build.cmd and buildandtest.cmd like Roslyn OR add a new buildonly command script/argument to the existing script so we can avoid running unit tests unnecessarily.

**Comment by 333fred** [comment]
:+1:

**Comment by mavasani** [comment]
Thanks @333fred.

The Debug/Release queues seem to be stuck, but are unaffected by this change. I am merging this in so I can play around with VSI runs. I'll address any further feedback with a separate change.

**Pull request merged**